src/libpmemfile/preload.c

```diff
@@ -229,6 +230,34 @@ log_write(const char *fmt, ...)
 	syscall_no_intercept(SYS_write, log_fd, buf, len);
 }
 
+void
+exit_with_msg(int ret, const char *msg)
+{
+	if (msg && msg[0] == '!') {
+		char buf[100];
+		char *b = strerror_r(errno, buf, sizeof(buf));
+		if (b == NULL) {
+			snprintf(buf, sizeof(buf), "Unknown errno %d", errno);
+		} else if (b != buf) {
+			strncpy(buf, b, sizeof(buf));
+			buf[sizeof(buf) - 1] = 0;
+		}
+
+		fprintf(stderr, "%s: %s\n", msg + 1, buf);
```

Comment:
If we already use snprintf,
what is the point of doing an `fprintf` and then an `fflush`?
Why not just have a `write(2, ...)`?